src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -193,6 +196,19 @@ export class EditDataGridPanel extends GridParentComponent {
 			}
 			// Store the value that was set
 			self.currentEditCellValue = event.item[event.cell];
+
+			// In case the last row is entered in, we need to wait to get the cell value after edit end
+			// so that we can add a row, submit the value and move down. (SlickGrid tries to go down immediately
+			// which fails, as we haven't added the new row yet).
+			if (self.isLastRowEnter) {
+				self.isLastRowEnter = false;
+				self.submitCurrentCellChange((result: EditUpdateCellResult) => {
+					self.table.grid.navigateDown();
+				},
+					(error: any) => {
+						this.notificationService.error(error);
```

Comment:
You shouldn't ever mix this and self - doing so at best is confusing and at worst is a bug. In this it should be capturing the this context so it's probably fine - but you should still use self to at least be consistent here.